compiler/transforms/schedule_memref_linalg.py

```diff
@@ -0,0 +1,302 @@
+from xdsl.context import MLContext
+from xdsl.dialects import builtin, memref_stream, stream
+from xdsl.ir import Block, Region
+from xdsl.ir.affine import AffineConstantExpr, AffineDimExpr, AffineExpr, AffineMap
+from xdsl.passes import ModulePass
+from xdsl.pattern_rewriter import (
+    PatternRewriter,
+    PatternRewriteWalker,
+    RewritePattern,
+    op_type_rewrite_pattern,
+)
+from xdsl.rewriter import InsertPoint
+
+from compiler.util.canonicalize_affine import canonicalize_map
+
+
+def disable_dims(map: AffineMap, dim: int) -> AffineMap:
+    """
+    Returns an affine map with the first `dim` dimensions set to 0
```

Comment:
wondering whether we should replace `first` with `leftmost`

Reply:
seems more clear yes